azconfig/client/src/main/java/com/azure/azconfig/AzConfigCredentialsPolicy.java

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.azconfig;
+
+import com.microsoft.rest.v3.http.HttpHeaders;
+import com.microsoft.rest.v3.http.HttpPipelineCallContext;
+import com.microsoft.rest.v3.http.HttpPipelineLogLevel;
+import com.microsoft.rest.v3.http.HttpPipelineOptions;
+import com.microsoft.rest.v3.http.HttpRequest;
+import com.microsoft.rest.v3.http.HttpResponse;
+import com.microsoft.rest.v3.http.NextPolicy;
+import com.microsoft.rest.v3.http.policy.HttpPipelinePolicy;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import reactor.core.publisher.Mono;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.Base64;
+
+/**
+ * Creates a policy that authenticates request with AzConfig service.
+ */
+public final class AzConfigCredentialsPolicy implements HttpPipelinePolicy {
+    private static final String SIGNED_HEADERS = "host;x-ms-date;x-ms-content-sha256";
+    private static final String KEY_VALUE_APPLICATION_HEADER = "application/vnd.microsoft.azconfig.kv+json";
+
+    private static final String HOST_HEADER = "Host";
+    private static final String DATE_HEADER = "x-ms-date";
+    private static final String CONTENT_HASH_HEADER = "x-ms-content-sha256";
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+    private static final String CONTENT_TYPE_HEADER = "Content-Type";
+    private static final String ACCEPT_HEADER = "Accept";
+
+    private final AzConfigClient.AzConfigCredentials credentials;
+    private final HttpPipelineOptions options;
+
+    /**
+     * Initializes a new instance of AzConfigCredentialsPolicy based on credentials.
+     *
+     * @param credentials for the Configuration Store in Azure
+     */
+    AzConfigCredentialsPolicy(AzConfigClient.AzConfigCredentials credentials) {
+        this(credentials, new HttpPipelineOptions(null));
+    }
+
+    /**
+     * Initializes a new instance of AzConfigCredentialsPolicy based on credentials.
+     *
+     * @param credentials for the Configuration Store in Azure
+     * @param options the request options
+     */
+    AzConfigCredentialsPolicy(AzConfigClient.AzConfigCredentials credentials, HttpPipelineOptions options) {
```

Comment:
Let's discuss HttpPipelineOptions. I don't think it will exist for long, if I get my way here: https://github.com/Azure/autorest-clientruntime-for-java/issues/557